packages/terra-date-picker/tests/wdio/date-picker-spec.js

```diff
@@ -672,6 +672,335 @@ Terra.describeViewports('Date Picker', ['medium'], () => {
+          browser.keys('-');
+
+          const currentDate = new Date();
+          currentDate.setDate(currentDate.getDate() - 1);
+          expect($('input')).toHaveValue(currentDate.toISOString().split('T')[0]);
+        });
+
+        it('sets Date to yesterday if on year input', () => {
+          Terra.hideInputCaret('input[name="terra-date-year-date-input"]');
+          $('input[name="terra-date-year-date-input"]').click();
+          browser.keys('-');
+
+          const currentDate = new Date();
+          currentDate.setDate(currentDate.getDate() - 1);
+          expect($('input')).toHaveValue(currentDate.toISOString().split('T')[0]);
+        });
+      });
+
+      describe('if partially filled', () => {
+        beforeEach(() => {
+          browser.url('/#/raw/tests/terra-date-picker/date-picker/date-picker-default');
+        });
+
+        it('sets Date to yesterday on month input', () => {
+          Terra.hideInputCaret('input[name="terra-date-month-date-input"]');
+          $('input[name="terra-date-month-date-input"]').click();
+          browser.keys('06');
+          $('input[name="terra-date-month-date-input"]').click();
+          browser.keys('-');
+
+          const currentDate = new Date();
+          currentDate.setDate(currentDate.getDate() - 1);
+          expect($('input')).toHaveValue(currentDate.toISOString().split('T')[0]);
+        });
+
+        it('sets Date to yesterday if on day input', () => {
+          Terra.hideInputCaret('input[name="terra-date-day-date-input"]');
+          $('input[name="terra-date-day-date-input"]').click();
+          browser.keys('20');
+          $('input[name="terra-date-day-date-input"]').click();
+          browser.keys('-');
+
+          const currentDate = new Date();
+          currentDate.setDate(currentDate.getDate() - 1);
+          expect($('input')).toHaveValue(currentDate.toISOString().split('T')[0]);
+        });
+
+        it('sets Date to yesterday if on year input', () => {
+          Terra.hideInputCaret('input[name="terra-date-year-date-input"]');
+          $('input[name="terra-date-year-date-input"]').click();
+          browser.keys('2005');
+          $('input[name="terra-date-year-date-input"]').click();
+          browser.keys('-');
+
+          const currentDate = new Date();
+          currentDate.setDate(currentDate.getDate() - 1);
+          expect($('input')).toHaveValue(currentDate.toISOString().split('T')[0]);
+        });
+      });
+
+      describe('if a valid date', () => {
+        beforeEach(() => {
+          browser.url('/#/raw/tests/terra-date-picker/date-picker/date-picker-start-date');
+        });
+
+        it('sets Date to valid date minus 1 day if pressed in month', () => {
+          Terra.hideInputCaret('input[name="terra-date-month-date-input"]');
+          $('input[name="terra-date-month-date-input"]').click();
+          browser.keys('-');
+
+          expect($('input')).toHaveValue('2017-03-31');
```

Comment:
How are you setting the initial condition of 2017-03-30?

Reply:
the URL has a date input with a starting value of `2017-04-01`
https://github.com/cerner/terra-framework/blob/5a4b1cf7be14410bd8173b971135be0edbba2ff9/packages/terra-date-picker/src/terra-dev-site/test/date-picker/DatePickerStartDate.test.jsx